flink-libraries/flink-ml/src/main/scala/org/apache/flink/ml/recommendation/ALS.scala

```diff
@@ -535,8 +581,17 @@ object ALS {
     itemOut: DataSet[(Int, OutBlockInformation)],
     userIn: DataSet[(Int, InBlockInformation)],
     factors: Int,
-    lambda: Double, blockIDPartitioner: FlinkPartitioner[Int]):
+    lambda: Double, blockIDPartitioner: FlinkPartitioner[Int],
+    implicitPrefs: Boolean,
+    alpha: Double):
   DataSet[(Int, Array[Array[Double]])] = {
+    // retrieve broadcast XtX matrix in implicit case
+    val XtXtoBroadcast = if (implicitPrefs) {
```

Comment:
I tried to fit the notation of the explicit ALS code, as it uses `userXtX` notation in the `updateFactors` function. I think it might be confusing to use two different notations is the code, even if the paper uses another notation.